File: backend/app/api/v1/endpoints/life_theme.py
"""
Life Theme API endpoints
"""

from fastapi import APIRouter, Depends, HTTPException, status
from sqlalchemy.orm import Session
from app.db.session import get_db
from app.core.security import get_current_user
from app.models.user import User
from app.models.life_theme import LifeTheme
from app.schemas.life_theme import LifeThemeCreate, LifeThemeUpdate, LifeThemeRead
from uuid import uuid4
from typing import Optional

router = APIRouter()


@router.get("/life-theme", response_model=Optional[LifeThemeRead])
def get_life_theme(
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    theme = (
        db.query(LifeTheme)
        .filter_by(user_id=str(current_user.id))
        .order_by(LifeTheme.updated_at.desc())
        .first()
    )
    return theme


@router.post(
    "/life-theme", response_model=LifeThemeRead, status_code=status.HTTP_201_CREATED
)
def set_life_theme(
    theme_in: LifeThemeCreate,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    # Only one active theme per user: create new, do not update previous
    theme = LifeTheme(
        id=str(uuid4()),
        user_id=str(current_user.id),
        theme_text=theme_in.theme_text,
    )
    db.add(theme)
    db.commit()
    db.refresh(theme)
    return theme

File: frontend/src/api/lifeTheme.ts
import axios from "./client";

export interface LifeTheme {
  id: string;
  theme_text: string;
  created_at: string;
  updated_at: string;
}

export async function getLifeTheme(): Promise<LifeTheme | null> {
  const res = await axios.get<LifeTheme | null>("/life-theme");
  return res.data;
}

export async function setLifeTheme(theme_text: string): Promise<LifeTheme> {
  const res = await axios.post<LifeTheme>("/life-theme", { theme_text });
  return res.data;
}
